src/seedu/addressbook/Main.java

```diff
@@ -78,15 +78,26 @@ private void exit() {
         System.exit(0);
     }
 
-    /** Reads the user command and executes it, until the user issues the exit command.  */
+    /** Reads the user command and executes it, until the user issues the exit command.
+     *
+     * I referenced PR #133 to use similar logic in handling the exceptions being thrown throughout the methods.
+     *
+     */
     private void runCommandLoopUntilExitCommand() {
         Command command;
         do {
             String userCommandText = ui.getUserCommand();
             command = new Parser().parseCommand(userCommandText);
-            CommandResult result = executeCommand(command);
-            recordResult(result);
-            ui.showResultToUser(result);
+
+            try {
+                CommandResult result = executeCommand(command);
+                recordResult(result);
+                ui.showResultToUser(result);
+            } catch (StorageOperationException e ) {
+                ui.showToUser(e.getMessage());
```

Comment:
As discussed in class, the control goes to showToUser method that prints a friendly error message. Is there something else we can do by handling exceptions more gracefully? Maybe save the program state to give the user a chance to change the file permission?